application/password_resets/password_resets_view.py

```diff
@@ -0,0 +1,64 @@
+"""
+Handles all logic of the password reset api
+"""
+from urllib.parse import urlencode, parse_qs, urlsplit, urlunsplit
+from flask import abort, render_template
+from flask_apispec import MethodResource, marshal_with, use_kwargs
+from flask_jwt_extended.exceptions import JWTDecodeError
+from jwt.exceptions import ExpiredSignatureError
+from flask_mail import Message
+from application.extensions import mail
+from application.responders import no_content
+from application.models import User
+from application.users.user_schema import USER_SCHEMA
+from application.password_resets import password_resets_validator, password_reset_jwt_manager
+
+
+class PasswordResets(MethodResource):
+    """Routes for resetting a password with a reset link."""
+
+    @use_kwargs(password_resets_validator.create_args())
+    def post(self, **params):
+        """Logic for sending a password reset link"""
+        user = User.find_by_email_or_username(params["email_or_username"])
+        if not user:
+            abort(404, "Could not find a user with that email address")
+        token = password_reset_jwt_manager.encode_reset_token(user)
+        reset_url = set_query_parameter(params["base_url"], "reset_token", token)
+        send_reset_password_email(reset_url, user)
+        return no_content()
```

Comment:
Should this really be a no content response? How about 201 (created reset), 200(just general okay) or 202 (Accepted request ready for processing).

Reply:
No Content is mostly just used for a Delete request. 201 seems suitable here.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201
I am not really creating a new resource (only if you think of the email as an resource) and I don't return anything. 201 should return something. Maybe a 200 would be okay here, but nothing says that 204 is reserved for delete

Reply:
discord uses 204 here as well. So I'm fine with it :D